### ISSUE TITLE
questions about label transform

### ISSUE THREAD

**Issue by JIBSN** [computational-cell-analytics/micro-sam#361]
Hi, 
I am now fintuning on livecell dataset using modified livelcell_finetuning.py script. I used default SAM architecture without unetr decoder. I was confused about label_transform parameter. I have reviewed the label.py in torch_em library, I found that there are three types of distances, i.e. distances, boundary_distances, directed_distances. I found that it can cause errors when calculating loss, input shape is not equal to target shape.  So, I want to know what's the purpose of doing label transform.  By the way, I can now train successfully with this, label_transform = PerObjectDistanceTransform(distances=True, boundary_distances=False, directed_distances=False, foreground=False, instances=False, min_size=25 ) . Is it correct?

**Comment by anwai98** [comment]
Hi @JIBSN,

The idea of `label_transform` parameter is to transform the labels to a desired representation. (for example: to get object boundaries, to get object distance maps, to binarize the instances, etc.).

For the new instance segmentation functionality in micro-sam, we compute the [distances](https://github.com/constantinpape/torch-em/blob/main/torch_em/transform/label.py#L284) per object (to the centers and boundaries), learn these objectives with our new instance segmentation feature, and perform [postprocessing](https://github.com/constantinpape/torch-em/blob/main/torch_em/util/segmentation.py#L122) on top to get the instance segmentation from the predicted distances.

If you wish to use micro-sam to fine-tune without UNETR decoder, you need not pass the `PerObjectDistanceTransform` to the `label_transform` (with a few more changes in other parts of the model, like the trainer choice, etc). See a modular example [here](https://github.com/computational-cell-analytics/micro-sam/tree/master/examples/finetuning).

**Comment by JIBSN** [comment]
Hi, @anwai98,
Reallly thanks for your quick reply, your answer indeed helped me. I will try the new instance segmentation soon. 
One more question, can the model be trained on NVIDAI 4090 card with 24GB memory?

**Comment by anwai98** [comment]
I can't say for sure, but I think it should fit on the mentioned resources, if you use the `vit_b` (the default) for fine-tuning and reduce the number of objects per batch used for finetuning (you can try to reduce it to ~20% of the current value) (see [here](https://github.com/computational-cell-analytics/micro-sam/blob/master/examples/finetuning/finetune_hela.py#L86)).

**Comment by anwai98** [comment]
Hi @JIBSN,

> can the model be trained on NVIDAI 4090 card with 24GB memory?

I did a little bit of exploration on this, and my thoughts are that `vit_b` should fit on the mentioned resource with the default parameters. If you get OOM issues, reducing the number of objects per batch to ~15 / 20 (default is 25) should do the trick. Hope it helps! :)
